autoload/clever_f.vim

```diff
@@ -255,7 +255,7 @@ function! clever_f#find(map, char_num) abort
     let moves_forward = s:moves_forward(before_pos, next_pos)
 
     " update highlight when cursor moves across lines
-    let mode = mode(1)
+    let mode = s:mode()
     if g:clever_f_mark_char
         if next_pos[0] != before_pos[0]
             \ || (a:map ==? 't' && !s:first_move[mode] && clever_f#compat#xor(s:moved_forward, moves_forward))
```

Comment:
An error occurred here.

For example:
When a user typed `dtx`, `clever_f#find_with()` is called and `let s:first_move[mode] = 1` is set.  `mode` is `no` here.
Next, this function is called via `v:<C-u>call clever_f#find(...)` in operator-pending mode, `mode(1)` returns `nov` because there is `v` prefix.  `s:first_move` doesn't have `nov` as key so `E716` occurs.

This is reproduced by a newly added test case.